docs/source/configuration.rst
Configuration
=============

In this section we will guide you through the options you need to configure
to setup your moocng instance.

Most of the settings go to a file called `local_settings.py` which is located
inside the moocng package. It is very important that you **never modify** the
`settings.py` file directly since it will be overriden when moocng is update
with a new version.

If you have followed the installation instructions, the local_settings.py file
should be located at `/var/www/moocng/lib/python2.7/site-packages/moocng-X.Y.Z-py2.7.egg/moocng/local_settings.py`

.. note::
  Bear in mind that the exact path may be different in your case, specially
  the Python and moocng version numbers. The path
  fragment :file:`moocng-X.Y.Z-py2.7` is ficticious and will be something like
  |full_release_name| in real life.


Database connection
-------------------

The first you should configure is the database connection settings. Actually
you probably have done that if you have followed the installation instructions
since this is required to create the database schema.

.. code-block:: python

 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.postgresql_psycopg2',
         'NAME': 'moocng',
         'USER': 'moocng',
         'PASSWORD': 'secret',
         'HOST': '',
         'PORT': '',
     }
 }


Rabbitmq connection
-------------------

Similar to the database connection settings, you need to configure
the message broker connection settings:

.. code-block:: python

 BROKER_URL = 'amqp://moocng:moocngpassword@localhost:5672/moocng'

Where the syntax of that url is `amqp://user:password@host:port/virtual_host`

FFmpeg location
---------------

It's required to configure the FFmpeg path in the moocng settings.
The default value is:

.. code-block:: python

  FFMPEG = '/usr/bin/ffmpeg'


Static root
-----------

In this step you will collect all necessary static resources needed by
moocng and put them in a single directory so you can serve them directly
through your web server increasing the efficiency of the whole system.

The STATIC_ROOT option is the absolute path where the static files will
be copied. It is recommended to be placed outside the moocng package.
The user who runs the Apache process will need read access to this directory.

.. code-block:: python

  STATIC_ROOT = '/var/www/moocng/static'

Once you have configured this option you should run the `collectstatic`
Django command to copy the static files to that destination:

.. code-block:: bash

  $ source /var/www/moocng/bin/activate
  $ django-admin.py collectstatic --settings=moocng.settings

 You have requested to collect static files at the destination
 location as specified in your settings file.

 This will overwrite existing files.
 Are you sure you want to do this?

 Type 'yes' to continue, or 'no' to cancel: yes

Media root
----------

In a similar fashion as the Static root, the media root is an
absolute path where the files uploaded by the moocng users will
be stored. It is also recommended to put this directory outside
the moocng package:

.. code-block:: python

  MEDIA_ROOT = '/var/www/moocng/media'

You should create this directory since you are going to change its
permissions when configuring the web server:

.. code-block:: bash

  $ mkdir /var/www/moocng/media


Secret key
----------

The secret key is a random string that Django uses in several places
like the CSRF attack protection. It is considered a security problem
if you don't change this value and leave it as the moocng default.

You can generate a random value with the following command:

.. code-block:: bash

  $ tr -c -d '0123456789abcdefghijklmnopqrstuvwxyz' </dev/urandom | dd bs=32 count=1 2>/dev/null;echo

And then set it in the SECRET_KEY option:

.. code-block:: python

  SECRET_KEY = 'qt6p480yug2t36on4ugpynp31tyveq39'


Google Analytics support
------------------------

This setting is optional and allows you to integrate your moocng with Google
Analytics so you can track who, when and how uses your site.

Just set the Google Analytics Code in this setting:

.. code-block:: python

  GOOGLE_ANALYTICS_CODE = 'XX-XXXX'


Theme
-----

TODO


Registry and profile URLs
-------------------------

moocng does not handle any user registration or user profile information.
It needs an external service for this. There are two options in the settings
to configure these urls:

.. code-block:: python

  REGISTRY_URL = 'https://idp.example.com/simplesaml/module.php/userregistration/newUser.php'
  PROFILE_URL = 'https://idp.example.com/simplesaml/module.php/userregistration/reviewUser.php'


Amazon S3 configuration
-----------------------

moocng use S3 to storage users uploaded files. You need an Amazon AWS account
and create a bucket to store the files.

The bucket must be configured with the next CORS configuration:

.. code-block:: xml

  <?xml version="1.0" encoding="UTF-8"?>
  <CORSConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
      <CORSRule>
          <AllowedOrigin>*</AllowedOrigin>
          <AllowedMethod>PUT</AllowedMethod>
          <MaxAgeSeconds>3000</MaxAgeSeconds>
          <AllowedHeader>Content-Type</AllowedHeader>
          <AllowedHeader>x-amz-acl</AllowedHeader>
          <AllowedHeader>origin</AllowedHeader>
          <AllowedHeader>Accept</AllowedHeader>
          <AllowedHeader>Accept-Charset</AllowedHeader>
          <AllowedHeader>Accept-Encoding</AllowedHeader>
          <AllowedHeader>Accept-Language</AllowedHeader>
          <AllowedHeader>Access-Control-Request-Headers</AllowedHeader>
          <AllowedHeader>Access-Control-Request-Method</AllowedHeader>
          <AllowedHeader>Connection</AllowedHeader>
          <AllowedHeader>Host</AllowedHeader>
          <AllowedHeader>Origin</AllowedHeader>
          <AllowedHeader>Referer</AllowedHeader>
          <AllowedHeader>User-Agent</AllowedHeader>
      </CORSRule>
  </CORSConfiguration>

To improve the security in production environments you can define a more strict
AllowedOrigin setting in your CORS configuration.

And your settings must define your account data, your bucket and the expire
time of upload permissions.

.. code-block:: python

  AWS_ACCESS_KEY_ID = "your-access-key-id"
  AWS_SECRET_ACCESS_KEY = "your-secret-key-id"
  AWS_STORAGE_BUCKET_NAME = "your-bucket-name"
  AWS_S3_UPLOAD_EXPIRE_TIME = (60 * 5) # 5 minutes


Web server configuration
------------------------

The recommended way to serve a moocng site is with a real web server that
supports the WSGI (Web Server Gateway Interface) protocol. This is no
surprise since the same applies to Django.

If you use the Apache web server all you need to do is write the
following configuration into your specific virtual host section:

.. code-block:: none

  NameVirtualHost *:80

  <VirtualHost *:80>
    ServerName moocng.example.com
    DocumentRoot /var/www/moocng/static/

    SetEnv VIRTUALENV /var/www/moocng

    WSGIScriptAlias / /var/www/moocng/lib/python2.7/site-packages/moocng-X.Y.Z-py2.7.egg/moocng/wsgi.py
    Alias /static/ /var/www/moocng/static/
    Alias /media/ /var/www/moocng/media/
  </VirtualHost>

.. note::
  Bear in mind that the exact path may be different in your case, specially
  the Python and moocng version numbers. The path
  fragment :file:`moocng-X.Y.Z-py2.7` is ficticious and will be something like
  |full_release_name| in real life.

Finally, you need to make sure that the user that the Apache run as has write
access to the media directory and read access to the static directory of your
moocng site.

.. code-block:: bash

  # Fedora example:
  $ chown apache:apache /var/www/moocng/media
  $ chown apache:apache /var/www/moocng/static

  # Debian/Ubuntu example:
  $ chown www-data:www-data /var/www/moocng/media
  $ chown www-data:www-data /var/www/moocng/static


SAML configuration
------------------

SAML require a cert. You can create your own self-signed certificates. For other purposes buy them:

  * Follow the first five steps of this guide: http://www.akadia.com/services/ssh_test_certificate.html
  * Create a directory called "saml2" at you moong folder
  * Create inside it a certs directory
  * Copy the 'attributemaps' of moocng inside the saml2
  * Copy server.key and server.crt to saml2/certs

.. code-block:: bash

  openssl genrsa -des3 -out server.key 2048
  openssl req -new -key server.key -out server.csr
  cp server.key server.key.org
  openssl rsa -in server.key.org -out server.key
  openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt

In moocng, in the settings.py there is a SAML_CONFIG var. You must copy this
var in your local_settings and configure the params based in your environment.

moocng uses djangosaml2, to config it check the doc at `http://pypi.python.org/pypi/djangosaml2 <http://pypi.python.org/pypi/djangosaml2>`_

In order to connect openmooc with an IdP, you will need its metadata. Download
it and save as remote_metadata.xml (check the saml configuration to check that
the path and name match)

Now you need to add the SAML SP metadata to your IdP. First of all you need to
configure in the IdP the metarefresh issue. After that you can go to the idp
and call update entries, You can go to a url like this:
https://idp.example.com/simplesaml/module.php/metarefresh/fetch.php


Asset configuration
-------------------

Slot duration time of assets should always be multiple of the asset slot
granularity.

That slot granularity is set to five minutes by default. To use another value,
simply specify a different value (in minutes) in the ASSET_SLOT_GRANULARITY
property of the settings file.


Enabling required services
--------------------------

There are several services that need to be active in the server so it is
best to configure them to start on the server boot.

.. code-block:: bash

  $ chkconfig --add httpd
  $ chkconfig --add rabbitmq-server
  $ chkconfig --add postgresql
  $ chkconfig --add mongod
  $ chkconfig --add celeryd
  $ chkconfig postgresql on
  $ chkconfig httpd on
  $ chkconfig rabbitmq-server on
  $ chkconfig mongod on
  $ chkconfig celeryd on

Openbadges integration
----------------------

The badges used in the platform can be integrated with Mozilla OpenBadges system (http://openbadges.org/). Every time, a student gets a badge, an assertion (OBI compliant) is created.

You can configure the openbadges service to use, by default the official backpack OpenBadges service:

.. code-block:: python

  BADGES_SERVICE_URL = 'backpack.openbadges.org'

To generate the assertion, we use some functions attached to post_save signals:

.. code-block:: python

  save_identity_for_user()

Copies the actual user data to the assertion for future consistency.

.. code-block:: python

  create_identity_for_user()

As the assertion identity hash contains the email information, if the email changes, the identity should also change.

You must define your issuer info in the settings variables.

.. code-block:: python

  BADGES_SERVICE_URL = "backpack.openbadges.org"
  BADGES_ISSUER_NAME = "OpenMOOC"
  BADGES_ISSUER_URL = "http://openmooc.org"
  BADGES_ISSUER_DESCRIPTION = ""
  BADGES_ISSUER_IMAGE = ""
  BADGES_ISSUER_EMAIL = ""


moocng/static/css/bootstrap-customization-options.txt
This is the customization we have done to Bootstrap.

See http://twitter.github.com/bootstrap/download.html

1. Choose components:

Scaffolding:
  - Normalize and reset
  - Body type and links
  - Grid system
  - Layouts

Base CSS:
  - Headings, body, etc
  - Labels and badges
  - Buttons
  - Icons

Components:
  - Buttons groups and dropdowns
  - Nav, tabs and pills
  - Alerts
  - Progress bars

JS Components:
  - Dropdowns
  - Collapse
  - Carousel

Miscellaneous
  - Wells
  - Close icon
  - Utilities
  - Component animations

Responsive:
  - Visible/hidden classes
  - Narrow tablets and below
  - Tablets to desktop
  - Large desktops

2. Select jQuery Plugins

- Transitions
- Dropdowns
- Togglable tabs
- Alert messages
- Buttons
- Collapse


moocng/settings.py
# Copyright 2012 Rooter Analysis S.L.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# Django settings for moocng project.

import os
BASEDIR = os.path.abspath(os.path.dirname(__file__))

DEBUG = True
TEMPLATE_DEBUG = DEBUG

FFMPEG_DEBUG = DEBUG

ADMINS = (
    ('Admin', 'admin@eopenmooc.org'),
)

MANAGERS = ADMINS

EMAIL_SUBJECT_PREFIX = '[OpenMOOC] | '

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',  # Add 'postgresql_psycopg2', 'mysql', 'sqlite3' or 'oracle'.
        'NAME': 'moocng',                # Or path to database file if using sqlite3.
        'USER': 'moocng',                # Not used with sqlite3.
        'PASSWORD': 'moocng',            # Not used with sqlite3.
        'HOST': '',                      # Set to empty string for localhost. Not used with sqlite3.
        'PORT': '',                      # Set to empty string for default. Not used with sqlite3.
    }
}

MONGODB_URI = 'mongodb://localhost:27017/moocng'

# Tastypie resource limit per page, 0 means unlimited
API_LIMIT_PER_PAGE = 0

#SMTP server
EMAIL_HOST = 'idp.openmooc.org'
SERVER_EMAIL = 'idp.openmooc.org'
DEFAULT_FROM_EMAIL = 'info@openmooc.org'
EMAIL_PORT = 25
EMAIL_HOST_USER = ''
EMAIL_HOST_PASSWORD = ''

EMAIL_SUBJECT_PREFIX = '[OpenMOOC] | '

# Amazon credentials
AWS_ACCESS_KEY_ID = ""
AWS_SECRET_ACCESS_KEY = ""
AWS_STORAGE_BUCKET_NAME = ""
AWS_S3_UPLOAD_EXPIRE_TIME = (60 * 5)  # 5 minutes

# Local time zone for this installation. Choices can be found here:
# http://en.wikipedia.org/wiki/List_of_tz_zones_by_name
# although not all choices may be available on all operating systems.
# On Unix systems, a value of None will cause Django to use the same
# timezone as the operating system.
# If running in a Windows environment this must be set to the same as your
# system time zone.
TIME_ZONE = 'Europe/Madrid'

# Language code for this installation. All choices can be found here:
# http://www.i18nguy.com/unicode/language-identifiers.html
LANGUAGE_CODE = 'en-us'

gettext = lambda s: s

LANGUAGES = (
    ('en', gettext('English')),
    ('es', gettext('Spanish')),
)

# the default value is 'django_language' but changing this
# to 'language' makes it easier to integrate with the IdP
LANGUAGE_COOKIE_NAME = 'language'

LOCALE_PATHS = (
    os.path.join(BASEDIR, 'locale'),
)

SITE_ID = 1

# If you set this to False, Django will make some optimizations so as not
# to load the internationalization machinery.
USE_I18N = True

# If you set this to False, Django will not format dates, numbers and
# calendars according to the current locale.
USE_L10N = True

# Use custom formats
FORMAT_MODULE_PATH = 'moocng.formats'

# If you set this to False, Django will not use timezone-aware datetimes.
USE_TZ = True

# Absolute filesystem path to the directory that will hold user-uploaded files.
# Example: "/home/media/media.lawrence.com/media/"
MEDIA_ROOT = os.path.join(BASEDIR, 'media')

# URL that handles the media served from MEDIA_ROOT. Make sure to use a
# trailing slash.
# Examples: "http://media.lawrence.com/media/", "http://example.com/media/"
MEDIA_URL = '/media/'

# Absolute path to the directory static files should be collected to.
# Don't put anything in this directory yourself; store your static files
# in apps' "static/" subdirectories and in STATICFILES_DIRS.
# Example: "/home/media/media.lawrence.com/static/"
STATIC_ROOT = os.path.join(BASEDIR, 'collected_static')

# URL prefix for static files.
# Example: "http://media.lawrence.com/static/"
STATIC_URL = '/static/'

# Additional locations of static files
STATICFILES_DIRS = (
    # Put strings here, like "/home/html/static" or "C:/www/django/static".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
    os.path.join(BASEDIR, 'static'),
)

# List of finder classes that know how to find static files in
# various locations.
STATICFILES_FINDERS = (
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
    'compressor.finders.CompressorFinder'
    # 'django.contrib.staticfiles.finders.DefaultStorageFinder',
)

# Make this unique, and don't share it with anybody.
SECRET_KEY = 'r$=%l$j4(#5a%$rd*g+q5o7!m3z&amp;b@z1+n*d!n2im-hf0n%730'

# List of callables that know how to import templates from various sources.
TEMPLATE_LOADERS = (
    'django.template.loaders.filesystem.Loader',
    'django.template.loaders.app_directories.Loader',
    # 'django.template.loaders.eggs.Loader',
)

MIDDLEWARE_CLASSES = (
    'django.middleware.common.CommonMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.locale.LocaleMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    # Uncomment the next line for simple clickjacking protection:
    # 'django.middleware.clickjacking.XFrameOptionsMiddleware',
)

ROOT_URLCONF = 'moocng.urls'

# Python dotted path to the WSGI application used by Django's runserver.
WSGI_APPLICATION = 'moocng.wsgi.application'

TEMPLATE_DIRS = (
    # Put strings here, like "/home/html/django_templates" or "C:/www/django/templates".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
    os.path.join(BASEDIR, 'templates'),
)

COMPRESS_CSS_FILTERS = [
    'compressor.filters.css_default.CssAbsoluteFilter',
    'compressor.filters.cssmin.CSSMinFilter',
]

COMPRESS_OFFLINE = False

INSTALLED_APPS = (
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.sites',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'grappelli',
    'django.contrib.admin',
    'django.contrib.flatpages',
    'adminsortable',
    'djcelery',
    'tinymce',
    'tastypie',
    'compressor',
    'moocng.contact',
    'moocng.badges',  # this must be defined before moocng.courses
    'moocng.courses',
    'moocng.assets',
    'moocng.portal',
    'moocng.videos',
    'moocng.teacheradmin',
    'moocng.enrollment',
    'moocng.api',
    'moocng.categories',
    'moocng.peerreview',
    'djangosaml2',
    'south',
    'django_mathjax',
    # Uncomment the next line to enable admin documentation:
    # 'django.contrib.admindocs',
    'moocng.media_contents',
)

# A sample logging configuration. The only tangible logging
# performed by this configuration is to send an email to
# the site admins on every HTTP 500 error when DEBUG=False.
# See http://docs.djangoproject.com/en/dev/topics/logging for
# more details on how to customize your logging configuration.
LOGGING = {
    'version': 1,
    'disable_existing_loggers': True,
    'formatters': {
        'verbose': {
            'format': '%(levelname)s %(asctime)s %(module)s %(process)d %(thread)d %(message)s'
        },
        'simple': {
            'format': '%(levelname)s %(message)s'
        },
    },
    'filters': {
        'require_debug_false': {
            '()': 'django.utils.log.RequireDebugFalse'
        },
    },
    'handlers': {
        'console': {
            'level': 'DEBUG',
            'class': 'logging.StreamHandler',
            'formatter': 'simple',
        },
        'mail_admins': {
            'level': 'ERROR',
            'class': 'django.utils.log.AdminEmailHandler',
            'filters': ['require_debug_false'],
        },
    },
    'loggers': {
        'django.request': {
            'handlers': ['console', 'mail_admins'],
            'level': 'DEBUG',
            'propagate': True,
        },
        'moocng.videos.download': {
            'handlers': ['console', 'mail_admins'],
            'level': 'INFO',
            'propagate': True,
        },
        'moocng.videos.tasks': {
            'handlers': ['console', 'mail_admins'],
            'level': 'INFO',
            'propagate': True,
        },
        'moocng.courses.admin': {
            'handlers': ['console', 'mail_admins'],
            'level': 'INFO',
            'propagate': True,
        },
    }
}

TEMPLATE_CONTEXT_PROCESSORS = (
    'django.contrib.auth.context_processors.auth',
    'django.core.context_processors.debug',
    'django.core.context_processors.i18n',
    'django.core.context_processors.media',
    'django.core.context_processors.static',
    'django.core.context_processors.tz',
    # 'django.core.context_processors.request',
    'django.contrib.messages.context_processors.messages',
    'moocng.context_processors.site',
    'moocng.context_processors.theme',
    'moocng.context_processors.extra_settings',
    'moocng.context_processors.idp_urls',
    'moocng.context_processors.google_analytics',
    'moocng.context_processors.certificate_url',
)

AUTHENTICATION_BACKENDS = (
    'django.contrib.auth.backends.ModelBackend',
    'moocng.courses.backends.Saml2BackendExtension',
)

FIXTURE_DIRS = (
    os.path.join(BASEDIR, 'fixtures', 'django.contrib.flatpages'),
    os.path.join(BASEDIR, 'fixtures', 'django.contrib.auth'),
)

TINYMCE_DEFAULT_CONFIG = {
    'theme': 'advanced',
    'theme_advanced_toolbar_location': 'top',
    'theme_advanced_buttons1': 'bold,italic,underline,strikethrough,separator,link,unlink,separator,undo,redo,copy,paste,separator,cleanup,separator,bullist,numlist',
    'theme_advanced_buttons2': '',
    'theme_advanced_buttons3': '',
}


GOOGLE_ANALYTICS_CODE = ''

GRAVATAR_URL_PREFIX = '//www.gravatar.com/'

MOOCNG_THEME = {
    # 'cert_banner': u'',
}


ENABLED_COMUNICATIONS = (
    'feedback',
    'incidence',
    'rights',
    'unsubscribe',
    'others'
)


#SHOW_TOS = True

FFMPEG = '/usr/bin/ffmpeg'

# Let authenticated users create their own courses
ALLOW_PUBLIC_COURSE_CREATION = False

# Make this unique, and don't share it with anybody else than payment system
# Override this in local settings
USER_API_KEY = '123456789'

# A list with the slugs of the courses that use the old qualification system
# where the normal units counted
COURSES_USING_OLD_TRANSCRIPT = []

# Enrollment methods
ENROLLMENT_METHODS = (
    'moocng.enrollment.methods.FreeEnrollment',
)

# Celery settings
import djcelery
djcelery.setup_loader()
CELERY_CREATE_MISSING_QUEUES = True

BROKER_URL = 'amqp://moocng:moocngpassword@localhost:5672/moocng'

REGISTRY_URL = 'https://idp.openmooc.org/simplesaml/module.php/userregistration/newUser.php'
PROFILE_URL = 'https://idp.openmooc.org/simplesaml/module.php/userregistration/reviewUser.php'
CHANGEPW_URL = 'https://idp.openmooc.org/simplesaml/module.php/userregistration/changePassword.php'
ASKBOT_URL_TEMPLATE = 'https://questions.example.com/%s/'

CERTIFICATE_URL = 'http://example.com/idcourse/%(courseid)s/email/%(email)s'  # Example, to be overwritten in local settings

MASSIVE_EMAIL_BATCH_SIZE = 30

PEER_REVIEW_TEXT_MAX_SIZE = 5000  # in chars
PEER_REVIEW_FILE_MAX_SIZE = 5  # in MB
PEER_REVIEW_ASSIGNATION_EXPIRE = 24  # in hours

ASSET_SLOT_GRANULARITY = 5  # Slot time of assets should be a multiple of this value (in minutes)

SESSION_EXPIRE_AT_BROWSER_CLOSE = True
LOGIN_URL = '/saml2/login/'
LOGIN_REDIRECT_URL = '/'
LOGOUT_URL = '/saml2/logout/'
LOGOUT_REDIRECT_URL = '/'

FREE_ENROLLMENT_CONSISTENT = False

SAML_CREATE_UNKNOWN_USER = True
SAML_ATTRIBUTE_MAPPING = {
    'mail': ('username', 'email', ),
    'cn': ('first_name', ),
    'sn': ('last_name', ),
    'eduPersonAffiliation': ('groups', ),
}

import saml2

SAML_CONFIG = {
    # full path to the xmlsec1 binary programm
    'xmlsec_binary': '/usr/bin/xmlsec1',

    # your entity id, usually your subdomain plus the url to the metadata view
    'entityid': 'https://moocng.example.com/saml2/metadata/',

    # directory with attribute mapping
    'attribute_map_dir': os.path.join(BASEDIR, 'attributemaps'),

    # this block states what services we provide
    'service': {
        # we are just a lonely SP
        'sp': {
            'name': 'Moocng SP',
            'endpoints': {
                # url and binding to the assetion consumer service view
                # do not change the binding or service name
                'assertion_consumer_service': [
                    ('https://moocng.example.com/saml2/acs/', saml2.BINDING_HTTP_POST),
                ],
                # url and binding to the single logout service view
                # do not change the binding or service name
                'single_logout_service': [
                    ('https://moocng.example.com/saml2/ls/', saml2.BINDING_HTTP_REDIRECT),
                ],
            },

            # in this section the list of IdPs we talk to are defined
            'idp': {
                # we do not need a WAYF service since there is
                # only an IdP defined here. This IdP should be
                # present in our metadata

                # the keys of this dictionary are entity ids
                'https://idp.example.com/simplesaml/saml2/idp/metadata.php': {
                    'single_sign_on_service': {
                        saml2.BINDING_HTTP_REDIRECT: 'https://idp.example.com/simplesaml/saml2/idp/SSOService.php',
                    },
                    'single_logout_service': {
                        saml2.BINDING_HTTP_REDIRECT: 'https://idp.example.com/simplesaml/saml2/idp/SingleLogoutService.php',
                    },
                },
            },
        },
    },

    # where the remote metadata is stored
    'metadata': {
        'local': [os.path.join(BASEDIR, 'remote_metadata.xml')],
    },

    # set to 1 to output debugging information
    'debug': 1,

    # certificate
    'key_file': os.path.join(BASEDIR, 'mycert.key'),  # private part
    'cert_file': os.path.join(BASEDIR, 'mycert.pem'),  # public part

    # own metadata settings
    'contact_person': [
        {'given_name': 'Sysadmin',
         'sur_name': '',
         'company': 'Example CO',
         'email_address': 'sysadmin@example.com',
         'contact_type': 'technical'},
        {'given_name': 'Boss',
         'sur_name': '',
         'company': 'Example CO',
         'email_address': 'admin@example.com',
         'contact_type': 'administrative'},
    ],
    # you can set multilanguage information here
    'organization': {
        'name': [('Example CO', 'es'), ('Example CO', 'en')],
        'display_name': [('Example', 'es'), ('Example', 'en')],
        'url': [('http://www.example.com', 'es'), ('http://www.example.com', 'en')],
    },
}

MEDIA_CONTENT_TYPES = [
    {
        'id': 'youtube',
        'name': 'YouTube',
        'handler': 'moocng.media_contents.handlers.youtube.YoutubeMediaContentHandler',
        'can_get_last_frame': True,
    },
    {
        'id': 'vimeo',
        'name': 'Vimeo',
        'handler': 'moocng.media_contents.handlers.vimeo.VimeoMediaContentHandler',
        'can_get_last_frame': False,
    },
    {
        'id': 'scribd',
        'name': 'Scribd',
        'handler': 'moocng.media_contents.handlers.scribd.ScribdMediaContentHandler',
        'can_get_last_frame': False,
    },
    {
        'id': 'prezi',
        'name': 'Prezi',
        'handler': 'moocng.media_contents.handlers.prezi.PreziMediaContentHandler',
        'can_get_last_frame': False,
    },
]

CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.dummy.DummyCache',
    },
}

MATHJAX_ENABLED = False
MATHJAX_LOCAL_PATH = STATIC_URL + 'js/libs/mathjax'
MATHJAX_CONFIG_FILE = "TeX-AMS-MML_HTMLorMML"
MATHJAX_CONFIG_DATA = {
    "elements": ['false-id-to-not-proccess-by-default'],
    "tex2jax": {
        "inlineMath": [
            ['$', '$'],
            ['\\(', '\\)']
        ]
    }
}

try:
    from local_settings import *
except ImportError:
    pass
